app.ts

```diff
@@ -0,0 +1,326 @@
+		state.searchedIndex = (searchPage - 1) * state.recordsPerPage + searchPageIndex;
+		displayData(Math.max(state.searchedIndex - state.recordsPerPage + 1, 0), state.recordsPerPage);
+	} else {
+		const previousFirstRecordIndex = state.currentFirstRecordIndex;
+		currentPage = Math.ceil((state.currentFirstRecordIndex + 1) / state.recordsPerPage);
+		displayData(state.currentFirstRecordIndex, state.recordsPerPage);
+	}
+	$("#loader").hide();
+	$("#tableWrapper").show();
+	if (currentPage === totalPages) {
+		$("#nextPageButton").hide();
+	} else {
+		$("#nextPageButton").show();
+	}
+	if (state.currentFirstRecordIndex <= 1) {
+		$("#prevPageButton").hide();
+	} else {
+		$("#prevPageButton").show();
+	}
+}
+
+window.onload = async function () {
+	$("#loader").hide();
+	updateScreen();
+	fetchColumns();
+	inputHandling();
+	$(window).on("resize", debounce(async () => {
+		$("#tableWrapper").hide();
+		$("#loader").show();
+		updateScreen()
+			.then(async () => {
+				const totalRecCount = await totalRecords();
+				if (currentPage * state.recordsPerPage > totalRecCount - 1) {
+					const lastPage = Math.ceil(totalRecCount / state.recordsPerPage);
+					currentPage = lastPage;
+					state.currentFirstRecordIndex = (lastPage - 1) * state.recordsPerPage;
+					return displayData(state.currentFirstRecordIndex, state.recordsPerPage);
+				}
+			})
+			.then(() => {
+				$("#loader").hide();
+				$("#tableWrapper").show();
+			})
+			.catch((error) => {
+				throw (error);
+			});
+	}, 250))
+
+	$("#searchInput").on("keydown", (e) => {
+		if (e.key === "e" || e.key === "E" || e.key === "."
+			|| e.key === "+" || e.key === "-") {
+			e.preventDefault();
+		}
+	});
+
+	$("#searchForm").submit(async (e) => {
+		e.preventDefault();
+		const searchValue = $("#searchInput").val();
+		$("#tableWrapper").hide();
+		$("#loader").show();
+		state.searchedIndex = null;
+		searchMethod(searchValue)
+			.then(() => {
+				if (state.searchedIndex !== null) {
+					currentPage = Math.ceil((state.searchedIndex + 1) / state.recordsPerPage);
+					state.currentFirstRecordIndex = Math.max(state.searchedIndex - state.recordsPerPage + 1, 0);
+				};
+			})
+		$("#loader").hide();
+		$("#tableWrapper").show();
+	});
+
+	$("#prevPageButton").on("click", async () => {
+		if ($("#prevPageButton").hasClass("hidden")) {
+			return;
+		}
+		$("#prevPageButton").addClass("hidden");
+		state.searchedIndex = null;
+		const firstRecordOfCurrentPage = (currentPage - 1) * state.recordsPerPage;
+		if (firstRecordOfCurrentPage <= 16) {
+			currentPage = 1;
+			state.currentFirstRecordIndex = 1;
+		} else {
+			currentPage--;
+			state.currentFirstRecordIndex -= state.recordsPerPage;
+		}
+		const fromRecord = state.currentFirstRecordIndex;
+		$("#nextPageButton").hide();
+		$("#prevPageButton").hide();
+		$("#tableWrapper").hide();
+		$("#loader").show();
+		displayData(fromRecord, state.recordsPerPage)
+			.then(() => {
+				setTimeout(() => {
```

Comment:
You should not have a `setTimeout` here, Unless there is something wrong with `displayData`. Please remove it.